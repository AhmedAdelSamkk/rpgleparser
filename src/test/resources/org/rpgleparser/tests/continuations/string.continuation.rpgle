      /FREE
         BegSr  CrtWrkFil;
jh3
jh3         //eMail workfile)
jh3         Exec Sql
jh3             Drop table QTEMP/EmailFile1;
jh3
            Exec Sql
jh3            CREATE TABLE QTEMP/EmailFile1
                (EmailData CHAR (500) NOT NULL WITH DEFAULT);
jh3
jh3         ExSr LodHed;
jh3
jh1         //
jh3         Exec Sql
jh3             Drop table  QTEMP/EmailFile2;
jh3
            Exec Sql
               CREATE TABLE QTEMP/EmailFile2
                (EmailData CHAR (500) NOT NULL WITH DEFAULT);

jh3         Eval Output =
jh3          'OD Pro#, +
jh3          Pickup Date, +
jh3          Delivery Date, +
jh3          DR Processed, +
jh3          Final Date, +
jh3          Previous Revenue, +
jh3          Current Revenue, +
jh3          Note';
jh3         Exec Sql
jh3            Insert into EmailFile2  Values (:Output);
jh3
         EndSr;
      /END-FREE
