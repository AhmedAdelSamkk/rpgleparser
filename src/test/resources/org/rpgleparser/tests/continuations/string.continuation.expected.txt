/*===TOKENS===*/
DIRECTIVE                < />
DIR_FREE                 <FREE>
EOL                      <>
OP_BEGSR                 <BegSr>
ID                       <CrtWrkFil>
FREE_SEMI                <;>
HIDDEN:LEAD_WS5_Comments <jh3  >
HIDDEN:LEAD_WS5_Comments <jh3  >
HIDDEN:COMMENTS          <       //>
HIDDEN:COMMENTS_TEXT     <eMail workfile)>
HIDDEN:LEAD_WS5_Comments <jh3  >
EXEC_SQL                 <Exec Sql>
WORDS                    <jh3             Drop table QTEMP/EmailFile1>
';'                      <;>
HIDDEN:LEAD_WS5_Comments <jh3  >
EXEC_SQL                 <Exec Sql>
WORDS                    <jh3            CREATE TABLE QTEMP/EmailFile1>
WORDS                    <(EmailData CHAR (500) NOT NULL WITH DEFAULT)>
';'                      <;>
HIDDEN:LEAD_WS5_Comments <jh3  >
HIDDEN:LEAD_WS5_Comments <jh3  >
OP_EXSR                  <ExSr>
ID                       <LodHed>
FREE_SEMI                <;>
HIDDEN:LEAD_WS5_Comments <jh3  >
HIDDEN:LEAD_WS5_Comments <jh1  >
HIDDEN:COMMENTS          <       //>
HIDDEN:LEAD_WS5_Comments <jh3  >
EXEC_SQL                 <Exec Sql>
WORDS                    <jh3             Drop table  QTEMP/EmailFile2>
';'                      <;>
HIDDEN:LEAD_WS5_Comments <jh3  >
EXEC_SQL                 <Exec Sql>
WORDS                    <CREATE TABLE QTEMP/EmailFile2>
WORDS                    <(EmailData CHAR (500) NOT NULL WITH DEFAULT)>
';'                      <;>
HIDDEN:LEAD_WS5_Comments <jh3  >
OP_EVAL                  <Eval>
ID                       <Output>
'='                      <=>
HIDDEN:LEAD_WS5_Comments <jh3  >
StringLiteralStart       <'>
StringContent            <OD Pro#, >
StringContent            <Pickup Date, >
StringContent            <Delivery Date, >
StringContent            <DR Processed, >
StringContent            <Final Date, >
StringContent            <Previous Revenue, >
StringContent            <Current Revenue, >
StringContent            <Note>
StringLiteralEnd         <'>
FREE_SEMI                <;>
HIDDEN:LEAD_WS5_Comments <jh3  >
EXEC_SQL                 <Exec Sql>
WORDS                    <jh3            Insert into EmailFile2  Values (:Output)>
';'                      <;>
HIDDEN:LEAD_WS5_Comments <jh3  >
OP_ENDSR                 <EndSr>
FREE_SEMI                <;>
DIRECTIVE                < />
DIR_ENDFREE              <END-FREE>
EOL                      <>
/*===TREE===*/
(r
  (statement (directive ' /' (beginfree_directive FREE) ))
  (subroutine
    (begsr
      (freeBEGSR BegSr (identifier (free_identifier (idOrKeyword CrtWrkFil))) ;)
    )
    (statement
      (free (exec_sql Exec Sql jh3             Drop table QTEMP/EmailFile1 ;))
    )
    (statement
      (free
        (exec_sql
          Exec Sql
          jh3            CREATE TABLE QTEMP/EmailFile1
          (EmailData CHAR (500) NOT NULL WITH DEFAULT)
          ;
        )
      )
    )
    (statement
      (free
        (baseExpression
          (op (op_exsr ExSr (identifier (free_identifier (idOrKeyword LodHed)))))
        )
        ;
      )
    )
    (statement
      (free (exec_sql Exec Sql jh3             Drop table  QTEMP/EmailFile2 ;))
    )
    (statement
      (free
        (exec_sql
          Exec Sql
          CREATE TABLE QTEMP/EmailFile2
          (EmailData CHAR (500) NOT NULL WITH DEFAULT)
          ;
        )
      )
    )
    (statement
      (free
        (baseExpression
          (op
            (op_eval
              Eval
              (evalExpression
                (assignmentExpression
                  (simpleExpression (identifier (free_identifier (idOrKeyword Output))))
                  =
                  (expression
                    (literal
                      '
                      'OD Pro#, '
                      'Pickup Date, '
                      'Delivery Date, '
                      'DR Processed, '
                      'Final Date, '
                      'Previous Revenue, '
                      'Current Revenue, '
                      Note
                      '
                    )
                  )
                )
              )
            )
          )
        )
        ;
      )
    )
    (statement
      (free
        (exec_sql Exec Sql jh3            Insert into EmailFile2  Values (:Output) ;)
      )
    )
    (endsr (freeENDSR EndSr ;))
  )
  (statement (directive ' /' (endfree_directive END-FREE) ))
)
/*======*/